# Исследование надежности заемщиков


**Описание проекта** 

Заказчик — кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

**Описание данных**
* `children` — количество детей в семье
* `days_employed` — общий трудовой стаж в днях
* `dob_years` — возраст клиента в годах
* `education` — уровень образования клиента
* `education_id` — идентификатор уровня образования
* `family_status` — семейное положение
* `family_status_id` — идентификатор семейного положения
* `gender` — пол клиента
* `income_type` — тип занятости
* `debt` — имел ли задолженность по возврату кредитов
* `total_income` — ежемесячный доход
* `purpose` — цель получения кредита

**План работы:**

1. Первичное изучение исходного датасета.
2. Предобработка данных:
    Обработка пропусков в данных.
    Обработка аномальных значений. 
    Обработка некорректных типов данных.
    Обработка дубликатов.
    Категоризация данных по разным признакам.
3. Исследование данных:
    Поиск факторов, влияющих на возврат кредита в срок.
    Исследование причин появления пропусков в данных.
4. Формирование общего вывода по работе.

## Откройте таблицу и изучите общую информацию о данных

**Задание 1. Импортируйте библиотеку pandas. Считайте данные из csv-файла в датафрейм и сохраните в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd

try:
    data = pd.read_csv('/datasets/data.csv')
except:
    # здесь была ссылка на данные с сервера Яндекс

**Задание 2. Выведите первые 20 строчек датафрейма `data` на экран.**

In [2]:
data.head(20)

children  days_employed  dob_years            education  education_id  \
0          1   -8437.673028         42               высшее             0   
1          1   -4024.803754         36              среднее             1   
2          0   -5623.422610         33              Среднее             1   
3          3   -4124.747207         32              среднее             1   
4          0  340266.072047         53              среднее             1   
5          0    -926.185831         27               высшее             0   
6          0   -2879.202052         43               высшее             0   
7          0    -152.779569         50              СРЕДНЕЕ             1   
8          2   -6929.865299         35               ВЫСШЕЕ             0   
9          0   -2188.756445         41              среднее             1   
10         2   -4171.483647         36               высшее             0   
11         0    -792.701887         40              среднее             1   
12         0            NaN         65              среднее             1   
13         0   -1846.641941         54  неоконченное высшее             2   
14         0   -1844.956182         56               высшее             0   
15         1    -972.364419         26              среднее             1   
16         0   -1719.934226         35              среднее             1   
17         0   -2369.999720         33               высшее             0   
18         0  400281.136913         53              среднее             1   
19         0  -10038.818549         48              СРЕДНЕЕ             1   

       family_status  family_status_id gender income_type  debt  \
0    женат / замужем                 0      F   сотрудник     0   
1    женат / замужем                 0      F   сотрудник     0   
2    женат / замужем                 0      M   сотрудник     0   
3    женат / замужем                 0      M   сотрудник     0   
4   гражданский брак                 1      F   пенсионер     0   
5   гражданский брак                 1      M   компаньон     0   
6    женат / замужем                 0      F   компаньон     0   
7    женат / замужем                 0      M   сотрудник     0   
8   гражданский брак                 1      F   сотрудник     0   
9    женат / замужем                 0      M   сотрудник     0   
10   женат / замужем                 0      M   компаньон     0   
11   женат / замужем                 0      F   сотрудник     0   
12  гражданский брак                 1      M   пенсионер     0   
13   женат / замужем                 0      F   сотрудник     0   
14  гражданский брак                 1      F   компаньон     1   
15   женат / замужем                 0      F   сотрудник     0   
16   женат / замужем                 0      F   сотрудник     0   
17  гражданский брак                 1      M   сотрудник     0   
18    вдовец / вдова                 2      F   пенсионер     0   
19         в разводе                 3      F   сотрудник     0   

     total_income                                 purpose  
0   253875.639453                           покупка жилья  
1   112080.014102                 приобретение автомобиля  
2   145885.952297                           покупка жилья  
3   267628.550329              дополнительное образование  
4   158616.077870                         сыграть свадьбу  
5   255763.565419                           покупка жилья  
6   240525.971920                       операции с жильем  
7   135823.934197                             образование  
8    95856.832424                   на проведение свадьбы  
9   144425.938277                 покупка жилья для семьи  
10  113943.491460                    покупка недвижимости  
11   77069.234271       покупка коммерческой недвижимости  
12            NaN                         сыграть свадьбу  
13  130458.228857                 приобретение автомобиля  
14  165127.911772              покупка жилой недвижимости  
15  116820.904450 

**Задание 3. Выведите основную информацию о датафрейме с помощью метода `info()`.**

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


 **Вывод:**

Мы импортировали исходный датасет и ознакомились со свойствами таблицы, полученной от кредитного отдела банка. 

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведите количество пропущенных значений для каждого столбца. Используйте комбинацию двух методов.**

In [4]:
data.isna().sum()

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

**Задание 5. В двух столбцах есть пропущенные значения. Один из них — `days_employed`. Пропуски в этом столбце вы обработаете на следующем этапе. Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца `income_type`. Например, у человека с типом занятости `сотрудник` пропуск в столбце `total_income` должен быть заполнен медианным доходом среди всех записей с тем же типом.**

In [5]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == t), 'total_income'].median()

### Обработка аномальных значений

**Задание 6. В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. Для реальных данных это нормально. Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода `abs()`.**

In [6]:
data['days_employed'] = data['days_employed'].abs()

**Задание 7. Для каждого типа занятости выведите медианное значение трудового стажа `days_employed` в днях.**

In [7]:
data.groupby('income_type')['days_employed'].agg('median')

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

У двух типов (безработные и пенсионеры) получатся аномально большие значения. Исправить такие значения сложно, поэтому оставьте их как есть. Тем более этот столбец не понадобится вам для исследования.

**Задание 8. Выведите перечень уникальных значений столбца `children`.**

In [8]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5])

**Задание 9. В столбце `children` есть два аномальных значения. Удалите строки, в которых встречаются такие аномальные значения из датафрейма `data`.**

In [9]:
data = data[(data['children'] != -1) & (data['children'] != 20)]

**Задание 10. Ещё раз выведите перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.**

In [10]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5])

### Удаление пропусков (продолжение)

**Задание 11. Заполните пропуски в столбце `days_employed` медианными значениями по каждого типа занятости `income_type`.**

In [11]:
for t in data['income_type'].unique():
    data.loc[(data['income_type'] == t) & (data['days_employed'].isna()), 'days_employed'] = \
    data.loc[(data['income_type'] == t), 'days_employed'].median()

**Задание 12. Убедитесь, что все пропуски заполнены. Проверьте себя и ещё раз выведите количество пропущенных значений для каждого столбца с помощью двух методов.**

In [12]:
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Задание 13. Замените вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [13]:
data['total_income'] = data['total_income'].astype(int)

### Обработка дубликатов

**Задание 14. Обработайте неявные дубликаты в столбце `education`. В этом столбце есть одни и те же значения, но записанные по-разному: с использованием заглавных и строчных букв. Приведите их к нижнему регистру. Проверьте остальные столбцы.**

In [14]:
data['education'] = data['education'].str.lower()

**Задание 15. Выведите на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалите их.**

In [15]:
data.duplicated().sum()

71

In [16]:
data = data.drop_duplicates()

### Категоризация данных

**Задание 16. На основании диапазонов, указанных ниже, создайте в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучателю с доходом 25000 нужно назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используйте собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [17]:
def categorize_income(income):
    try:
        if 0 <= income <= 30000:
            return 'E'
        elif 30001 <= income <= 50000:
            return 'D'
        elif 50001 <= income <= 200000:
            return 'C'
        elif 200001 <= income <= 1000000:
            return 'B'
        elif income >= 1000001:
            return 'A'
    except:
        pass

In [18]:
data['total_income_category'] = data['total_income'].apply(categorize_income)

**Задание 17. Выведите на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [19]:
data['purpose'].unique()

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

**Задание 18. Создайте функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используйте собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [20]:
def categorize_purpose(row):
    try:
        if 'автом' in row:
            return 'операции с автомобилем'
        elif 'жил' in row or 'недвиж' in row:
            return 'операции с недвижимостью'
        elif 'свад' in row:
            return 'проведение свадьбы'
        elif 'образов' in row:
            return 'получение образования'
    except:
        return 'нет категории'

In [21]:
data['purpose_category'] = data['purpose'].apply(categorize_purpose)

### Шаг 3. Исследуйте данные и ответьте на вопросы

#### 3.1 Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [22]:
# Выведем снова датасет с новыми столбцами total_income_category и purpose_category
data.head()

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы

In [23]:
# Найдем долю кредиторов, имеющих задолженность, при разном количестве детей
data_children_debt = data.groupby('children').agg({'debt': ['count', 'sum', 'mean']})
data_children_debt = data_children_debt.rename(columns={
    'count': 'Число заемщиков', 'sum': 'Число должников', 'mean': 'Средний % просрочки'
})
data_children_debt

debt                                    
         Число заемщиков Число должников Средний % просрочки
children                                                    
0                  14091            1063            0.075438
1                   4808             444            0.092346
2                   2052             194            0.094542
3                    330              27            0.081818
4                     41               4            0.097561
5                      9               0            0.000000

 **Вывод:**

Опишем полученные выше результаты:
* Около 7.5% заемщиков, у которых *нет детей*, имеют задолженность по возврату кредитов.
* Среди заемщиков, у которых *1-2 ребенка*, в среднем 9.2-9.5% имели задолженности по возврату кредитов. Между этими двумя группами (1 ребенок и 2 ребенка) нет существенной разницы. 
* Среди заемщиков с *3 детьми* около 8.2% имели задолженности по возврату кредитов.
* Для заемщиков с *4 детьми* доля заемщиков, имевших задолженность, наибольшая – 9.8%. 
* Заметим, что заемщики, имеющие *5 детей*, в среднем не имели задолженностей по возврату кредитов. 

Проанализировав доли кредиторов, имевших задолженности по возврату кредитов, при разном количестве детей, можно сделать вывод о том, что зависимость между числом детей и возвратом кредита в срок наблюдается.

#### 3.2 Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [24]:
# Найдем долю кредиторов, имеющих задолженность, при разном семейном статусе

data_family_status_debt = data.groupby('family_status').agg({'debt': ['count', 'sum', 'mean']})
data_family_status_debt = data_family_status_debt.rename(columns={
    'count': 'Число заемщиков', 'sum': 'Число должников', 'mean': 'Средний % просрочки'
})
data_family_status_debt

debt                                    
                      Число заемщиков Число должников Средний % просрочки
family_status                                                            
Не женат / не замужем            2796             273            0.097639
в разводе                        1189              84            0.070648
вдовец / вдова                    951              63            0.066246
гражданский брак                 4134             385            0.093130
женат / замужем                 12261             927            0.075606

Поскольку среди средних значений встречаются близкие значения, а также с учетом несбалансированности выборок, сгруппируем заемщиков по семейному статусу следующим образом:
1. Объединим в одну группу *(группа 1: не состоящие в официальном браке)* неженатых, незамужних и состоящих в гражданском браке, поскольку доля заемщиков, имеющих проблему с возвратом кредита в срок, составляет 9.3-9.8%. Данные группы можно логически объединить в том числе и по причине того, что гражданский брак официально не является браком и на юридическом уровне данная категория совпадает с категорией "не женат / не замужем".
2. Объединим в другую группу *(группа 2: не состоявщие в браке, но состоявшие ранее)* вдовцов, вдов и людей в разводе, поскольку доля заемщиков, имеющих проблему с возвратом кредита в срок, составляет 6.6-7.1%. Данные группы можно логически объединить в том числе и потому, что ранее данные заемщики состояли в браке, а на данный момент не состоят в браке.
3. В качестве третьей группы *(группа 3: состоящие в официальном браке)* оставим лиц, состоящих в официальном браке - "женат / замужем".

In [25]:
# Создадим функцию categorize_family_status, 
# которая будет определять группу в соответствии с семейным статусом
def categorize_family_status(family_status):
    if (family_status == 'Не женат / не замужем') or (family_status == 'гражданский брак'):
        return 'не состоящие в официальном браке'
    elif (family_status == 'вдовец / вдова') or (family_status == 'в разводе'):
        return 'не состоящие в браке, но состоявшие ранее'
    return 'состоящие в официальном браке'

In [26]:
# Добавим в таблицу data новый столбец family_status_official, 
# применив функцию categorize_family_status к столбцу family_status 
data['family_status_official'] = data['family_status'].apply(categorize_family_status)

In [27]:
# Найдем долю кредиторов, имеющих задолженность, при разном официальном семейном статусе 

data_family_status_official_debt = data.groupby('family_status_official').agg({'debt': ['count', 'sum', 'mean']})
data_family_status_official_debt = data_family_status_official_debt.rename(columns={
    'count': 'Число заемщиков', 'sum': 'Число должников', 'mean': 'Средний % просрочки'
})
data_family_status_official_debt

debt                  \
                                          Число заемщиков Число должников   
family_status_official                                                      
не состоящие в браке, но состоявшие ранее            2140             147   
не состоящие в официальном браке                     6930             658   
состоящие в официальном браке                       12261             927   

                                                               
                                          Средний % просрочки  
family_status_official                                         
не состоящие в браке, но состоявшие ранее            0.068692  
не состоящие в официальном браке                     0.094949  
состоящие в официальном браке                        0.075606

**Вывод:** 

Опишем полученные результаты:
1. Кредиторы, не состоящие в браке, но состоявшие ранее, **реже** остальных не возвращают кредиты в срок, и являются наиболее благонадежной группой заемщиков. Среди них около **6.9%** заемщиков не возвращает кредиты в срок.
2. Люди, не состоящие в официальном браке, **чаще** остальных не возвращают кредиты в срок, и являются наименее благонадежной группой заемщиков. Среди них около **9.5%** заемщиков не возвращают кредиты в срок. 
3. Среди людей, состоящих в официальном браке, около **7.6%** заемщиков имеют проблемы с возвратом кредита в срок. 

Проанализировав доли кредиторов, имевших задолженности по возврату кредитов, при разном семейном положении, можно сделать вывод о том, что зависимость между семейным положением и возвратом кредита в срок наблюдается.

#### 3.3 Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [28]:
data_income_debt = data.groupby('total_income_category').agg({'debt': ['count', 'sum', 'mean']})
data_income_debt = data_income_debt.rename(columns={
    'count': 'Число заемщиков', 'sum': 'Число должников', 'mean': 'Средний % просрочки'
})
data_income_debt

debt                                    
                      Число заемщиков Число должников Средний % просрочки
total_income_category                                                    
A                                  25               2            0.080000
B                                5014             354            0.070602
C                               15921            1353            0.084982
D                                 349              21            0.060172
E                                  22               2            0.090909


* Заметим, что наименее надежными заемщиками, исходя из полученных результатов являются заемщики *категории E*, то есть заемщики с наименьшим доходом (доход до 30 тыс.). Среди них **9.1%** заемщиков не возвращают кредиты в срок. 
* *Категория D* при этом является наиболее благонадежной, поскольку кредиторы из данной категории реже остальных в процентном соотношении имеют проблемы с возвратом кредита в срок. Всего **6%** таких заемщиков не возвращают кредиты в срок.
* Наименее надежными можно назвать *категории A и C* - около **8-8.5%** заемщиков данных категорий не возвращают кредиты в срок.
* Около **7.1%** заемщиков *категории B* не возвращают кредиты в срок.

Можно объяснить такие результаты следуюшим образом:
1. Кредиторы с доходом ниже 30 тыс. (категория E) действительно являются наименее надежными в силу небольшого дохода.
2. Кредиторы с доходом в диапазоне 30-50 тыс. (категория D) являются наиболее надежными, вероятно, потому что, имея небольшой доход, они стараются не ухудшить свой кредитный рейтинг и выплачивать всё в срок.
3. Кредиторы с доходами выше 50 тыс. (категории C, B и A) имеют среднюю надежность.

Таким образом, можно разделить кредиторов на три группы в соответствии с описаниием выше. Так банку в дальнейшем будет проще формировать скоринговую модель. 

In [29]:
# Создадим функцию categorize_income_for_scoring, 
# которая будет определять группу в соответствии с уровнем дохода:
def categorize_income_for_scoring(income):
    if income <= 30000:
        return 'низкий доход'
    elif income <= 50000:
        return 'средний доход'
    return 'хороший доход'

In [30]:
# Добавим в таблицу data новый столбец income_category_for_scoring, 
# применив функцию categorize_income_for_scoring к столбцу total_income 
data['income_category_for_scoring'] = data['total_income'].apply(categorize_income_for_scoring)

In [31]:
# Найдем долю кредиторов, имеющих задолженность, 
# при разном уровне доходов из столбца income_category_for_scoring

data_income_scoring_debt = data.groupby('income_category_for_scoring').agg({'debt': ['count', 'sum', 'mean']})
data_income_scoring_debt = data_income_scoring_debt.rename(columns={
    'count': 'Число заемщиков', 'sum': 'Число должников', 'mean': 'Средний % просрочки'
})
data_income_scoring_debt

debt                  \
                            Число заемщиков Число должников   
income_category_for_scoring                                   
низкий доход                             22               2   
средний доход                           349              21   
хороший доход                         20960            1709   

                                                 
                            Средний % просрочки  
income_category_for_scoring                      
низкий доход                           0.090909  
средний доход                          0.060172  
хороший доход                          0.081536

**Вывод:** 

Заметим, что число наблюдений в группе с низким доходом крайне низкая – всего 22 заемщика, поэтому рассматривать данные результаты при формировании скоринговой модели будет несколько некорректно.
При этом при формировании скоров можно учитывать зависимости, обнаруженные для групп со средним и хорошим доходом.

Проанализировав доли кредиторов, имевших задолженности по возврату кредитов, при разном уровне доходов, можно сделать вывод о том, что зависимость между уровнем доходов и возвратом кредита в срок наблюдается.

#### 3.4 Как разные цели кредита влияют на его возврат в срок?

In [32]:
data_purpose_debt = data.groupby('purpose_category').agg({'debt': ['count', 'sum', 'mean']})
data_purpose_debt = data_purpose_debt.rename(columns={
    'count': 'Число заемщиков', 'sum': 'Число должников', 'mean': 'Средний % просрочки'
})
data_purpose_debt

debt                                    
                         Число заемщиков Число должников Средний % просрочки
purpose_category                                                            
операции с автомобилем              4279             400            0.093480
операции с недвижимостью           10751             780            0.072551
получение образования               3988             369            0.092528
проведение свадьбы                  2313             183            0.079118

**Вывод:** 
- Заемщики, берущие кредиты для операций с автомобилем и для получения образования, в среднем одинаково себя ведут в плане возврата кредита в срок. Около таких 9.3% заемщиков имеют проблемы с возвратом кредитов в срок. Это наименее надежные заемщики.
- Заемщики, берущие кредиты для проведения свадьбы, имеют средний уровень надежности. около 7.9% таких заемщиков не возвращают кредиты в срок.
- Наиболее надежными являются заемщики, берущие кредит для операций с недвижимостью. Это можно объяснить как правило большой суммой кредита (заемщики более ответственно относятся к бОльшим суммам кредита). Также это можно объяснить льготными процентами по кредитам с недвижимостью.

#### 3.5 Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* 

В исходных данных пропуски встречались в следующих столбцах: `days_employed` и `total_income`. 

Для столбца `days_employed` причины появления пропусков могут быть следующими:
- Была утеряна трудовая книжка и некоторые данные не удалось восстановить, поэтому стаж в днях неизвестен.
- Среди опыта работы есть практики, стажировки, которые официально не оформлялись, следовательно, стаж в днях неизвестен.
- Неофициальное трудоустройство заемщика.

Для столбца `total_income` причины появления пропусков могут быть следующими:
- Наличие пассивного дохода, который трудно учесть (проценты по акциям, облигациям, сдача недвижимости в аренду и проч.)
- Ежемесячные выплаты включают в себя процент (с продаж), который разнится от месяца к месяцу, следовательно, точная сумма ежемесячного дохода неизвестна.
- Заемщик может быть иждивенец (женщина в декрете, живущая за счет супруга; студент, живущий за счет родителей), который не может указать точного дохода.

Все описанные выше причины относятся к человеческому фактору. Также стоит учесть возможность возникновения пропусков в данных в связи с техническими сбоями в работе систему учета банка.

#### 3.6 Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* 

Использование медианных значений при заполнении пропусков в количественных переменных является наилучшим решением, поскольку при наличии "выбросов" (сильно выделяющихся значений, которые редко встречаются в выборке) в данных медиана лучше показывает адекватную ситуацию. 

Рассмотрим, например, заработные платы в следующем коллективе: 100 т.р. у 5 сотрудников и 1 млн у ген.директора. **Медианная зарплата** будет составлять **100 т.р.**, в то время как **средняя зарплата** – **250 т.р**. При ежегодной индексации зарплат на 10% (110 т.р. у 5 сотрудников и 1.1 млн у гендира) **средняя зарплата вырастет** до 275 т.р. (**на 25 т.р.**), в то время как **медианная з.п. вырастет на 10 т.р**. 

Таким образом, при использовании среднего может ошибочно сложиться впечатление о том, что:
- во-первых, средняя з.п. в коллективе 250 т.р.,
- во-вторых, в среднем з.п. каждого сотрудника за год выросла на 25 т.р.

В то же самое время медиана адекватно показывает, что:
- во-первых, медианная з.п. в коллективе 100 т.р.,
- во-вторых, в среднем з.п. каждого сотрудника за год выросла на 10 т.р.

Поэтому заполнение пропусков медианным значением — лучшее решение для количественных переменных.

### Шаг 4: общий вывод.

В данных, полученных от банка, наблюдаются следующие зависимости:

1. Зависимость между числом детей и возвратом кредита в срок:
    * Заемщики, имеющие *5 детей*, в среднем не имели задолженностей по возврату кредитов. Они являются наиболее благонадежной группой заемщиков.
    * Следующими по надежности выступают заемщики, которых нет детей. Около **7.5%** заемщиков, у которых *нет детей*, имеют задолженность по возврату кредитов.
    * Среди заемщиков с *3 детьми* около **8.2%** имели задолженности по возврату кредитов.
    * Среди заемщиков, у которых *1-2 ребенка*, в среднем **9.2-9.5%** имели задолженности по возврату кредитов. Между этими двумя группами (1 ребенок и 2 ребенка) нет существенной разницы. 
    * Для заемщиков с *4 детьми* доля заемщиков, имевших задолженность, наибольшая – **9.8%**. 


2. Зависимость между семейным положением и возвратом кредита в срок:
Рекомендуется разделить кредиторов на следующие три группы по семейному положению: не состоящие в браке, но состоявшие ранее; не состоящие в официальном браке; состоящие в официальном браке.
    * **Не состоящие в браке, но состоявшие ранее** реже остальных не возвращают кредиты в срок, и являются **наиболее благонадежной** группой заемщиков. 
    * **Не состоящие в официальном браке** чаще остальных не возвращают кредиты в срок, и являются **наименее надежной** группой заемщиков. 
    * **Состоящие в официальном браке** имеют **средний уровень надежности**.
    
    
3. Зависимость между уровнем доходов и возвратом кредита в срок.
Рекомендуется разделить кредиторов на следующие три группы по уровню дохода: с доходом ниже 30 тыс.; с доходом в диапазоне 30-50 тыс.; с доходами выше 50 тыс.
    * Кредиторы **с доходом ниже 30 тыс.** являются **наименее надежными** в силу небольшого дохода.
    * Кредиторы **с доходом в диапазоне 30-50 тыс.** являются **наиболее надежными**.
    * Кредиторы **с доходами выше 50 тыс.** имеют **среднюю надежность**.
    
    
4. Зависимость между целью кредита и возвратом кредита в срок.
    * Заемщики, берущие кредиты **для операций с автомобилем** и **для получения образования** – **наименее надежные** заемщики.
    * Заемщики, берущие кредиты **для проведения свадьбы** имеют **средний уровень надежности**. 
    * **Наиболее надежными** являются заемщики, берущие кредит **для операций с недвижимостью**. 

Все полученные данные можно использовать при формировании кредитного скоринга заемщиков банком. 